Java Web Archive of release 3.5.9 won't deploy since included PrimeFaces and JSF versions are not compatible

I downloaded the latest stable release 3.5.9 as a Java Web Archive and tried to deploy it under Apache Tomcat 9.0.97 but it failed with:

```
10-Jan-2025 11:50:44.272 SEVERE [main] com.sun.faces.config.ConfigureListener.contextInitialized Critical error during deployment: 
	com.sun.faces.config.ConfigurationException: Factory 'javax.faces.context.FacesContextFactory' was not configured properly.
		at com.sun.faces.config.processor.FactoryConfigProcessor.verifyFactoriesExist(FactoryConfigProcessor.java:330)
		at com.sun.faces.config.processor.FactoryConfigProcessor.process(FactoryConfigProcessor.java:236)
		at com.sun.faces.config.ConfigManager.initialize(ConfigManager.java:439)
		at com.sun.faces.config.ConfigureListener.contextInitialized(ConfigureListener.java:227)
		at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4018)
		at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4460)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:599)
		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:571)
		at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:603)
		at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:1014)
		at org.apache.catalina.startup.HostConfig$DeployWar.run(HostConfig.java:1866)
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
		at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:123)
		at org.apache.catalina.startup.HostConfig.deployWARs(HostConfig.java:816)
		at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:468)
		at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1584)
		at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:312)
		at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:109)
		at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:389)
		at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:336)
		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:776)
		at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:721)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1203)
		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1193)
		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
		at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:145)
		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:749)
		at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:211)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.core.StandardService.startInternal(StandardService.java:415)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:874)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.startup.Catalina.start(Catalina.java:735)
		at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
		at java.base/java.lang.reflect.Method.invoke(Method.java:580)
		at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:345)
		at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:473)
	Caused by: javax.faces.FacesException: org.primefaces.context.PrimeFacesContextFactory
		at javax.faces.FactoryFinder.getImplGivenPreviousImpl(FactoryFinder.java:710)
		at javax.faces.FactoryFinder.getImplementationInstance(FactoryFinder.java:572)
		at javax.faces.FactoryFinder.access$500(FactoryFinder.java:140)
		at javax.faces.FactoryFinder$FactoryManager.getFactory(FactoryFinder.java:1120)
		at javax.faces.FactoryFinder.getFactory(FactoryFinder.java:379)
		at com.sun.faces.config.processor.FactoryConfigProcessor.verifyFactoriesExist(FactoryConfigProcessor.java:328)
		... 42 more
	Caused by: java.lang.reflect.InvocationTargetException
		at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:74)
		at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
		at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
		at javax.faces.FactoryFinder.getImplGivenPreviousImpl(FactoryFinder.java:690)
		... 47 more
	Caused by: java.lang.NoSuchMethodError: 'void javax.faces.context.FacesContextFactory.<init>(javax.faces.context.FacesContextFactory)'
		at org.primefaces.context.PrimeFacesContextFactory.<init>(PrimeFacesContextFactory.java:37)
		at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
		... 50 more
```

The Java Web Archive of release 3.5.9 currently includes `WEB-INF/lib/jsf-impl-2.2.20.jar`, `WEB-INF/lib/primefaces-14.0.6.jar` and `WEB-INF/lib/primefaces-extensions-14.0.6.jar`. But since the [PrimeFaces migration guide 13.0.0 -> 14.0.0](https://primefaces.github.io/primefaces/15_0_0/#/../migrationguide/14_0_0) says _JSF 2.0, 2.1 and 2.2 support is has been removed,_ the PrimeFaces and JSF versions in the Java Web Archive of release 3.5.9 are not compatible. For me, the solution was as simple as replacing `WEB-INF/lib/primefaces-14.0.6.jar` and `WEB-INF/lib/primefaces-extensions-14.0.6.jar` by `WEB-INF/lib/primefaces-13.0.6.jar` and `WEB-INF/lib/primefaces-extensions-13.0.6.jar` from the Java Web Archive of release 3.5.8.

--- a/deegree-core/deegree-core-sqldialect/deegree-sqldialect-postgis/src/main/java/org/deegree/sqldialect/postgis/PostGISDialect.java
+++ b/deegree-core/deegree-core-sqldialect/deegree-sqldialect-postgis/src/main/java/org/deegree/sqldialect/postgis/PostGISDialect.java
@@ -61,7 +61,7 @@ import org.deegree.sqldialect.SortCriterion;
 import org.deegree.sqldialect.filter.AbstractWhereBuilder;
 import org.deegree.sqldialect.filter.PropertyNameMapper;
 import org.deegree.sqldialect.filter.UnmappableException;
-import org.postgis.PGboxbase;
+import net.postgis.jdbc.PGboxbase;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -207,7 +207,7 @@ public class PostGISDialect extends AbstractSQLDialect implements SQLDialect {
 		return env;
 	}
 
-	private org.deegree.geometry.primitive.Point buildPoint(org.postgis.Point p, ICRS crs) {
+	private org.deegree.geometry.primitive.Point buildPoint(net.postgis.jdbc.geometry.Point p, ICRS crs) {
 		double[] coords = new double[p.getDimension()];
 		coords[0] = p.getX();
 		coords[1] = p.getY();

--- a/deegree-services/deegree-services-wpvs/src/main/java/org/deegree/services/wpvs/io/db/PostgisBackend.java
+++ b/deegree-services/deegree-services-wpvs/src/main/java/org/deegree/services/wpvs/io/db/PostgisBackend.java
@@ -47,12 +47,12 @@ import org.deegree.geometry.Envelope;
 import org.deegree.rendering.r3d.persistence.RenderableStore;
 import org.deegree.workspace.ResourceMetadata;
 import org.deegree.workspace.Workspace;
-import org.postgis.Geometry;
-import org.postgis.LinearRing;
-import org.postgis.PGbox3d;
-import org.postgis.PGgeometry;
-import org.postgis.Point;
-import org.postgis.Polygon;
+import net.postgis.jdbc.geometry.Geometry;
+import net.postgis.jdbc.geometry.LinearRing;
+import net.postgis.jdbc.PGbox3d;
+import net.postgis.jdbc.PGgeometry;
+import net.postgis.jdbc.geometry.Point;
+import net.postgis.jdbc.geometry.Polygon;
 
 /**
  * The <code>PostgisBackend</code> class adds postgis specific methods to the model
@@ -85,7 +85,7 @@ public class PostgisBackend extends DBBackend<PGgeometry> {
 			}
 			Polygon pgPolygon = (Polygon) geom;
 			ICRS crs = CRSManager.getCRSRef("EPSG:" + pgPolygon.getSrid());
-			org.postgis.LinearRing ring = pgPolygon.getRing(0);
+			net.postgis.jdbc.geometry.LinearRing ring = pgPolygon.getRing(0);
 			Point min = ring.getPoint(0);
 			Point max = ring.getPoint(2);
 			double[] mi = null;
@@ -145,7 +145,7 @@ public class PostgisBackend extends DBBackend<PGgeometry> {
 			points[4] = new Point(minD[0], minD[1], minD[2]);
 		}
 		linRing[0] = new LinearRing(points);
-		org.postgis.Polygon pgPoly = new org.postgis.Polygon(linRing);
+		net.postgis.jdbc.geometry.Polygon pgPoly = new net.postgis.jdbc.geometry.Polygon(linRing);
 		pgPoly.setSrid(parseSRID(geometry.getCoordinateSystem()));
 		return new PGgeometry(pgPoly);
 

--- a/deegree-services/deegree-webservices/pom.xml
+++ b/deegree-services/deegree-webservices/pom.xml
@@ -278,12 +278,12 @@
     <dependency>
       <groupId>org.primefaces</groupId>
       <artifactId>primefaces</artifactId>
-      <version>14.0.10</version>
+      <version>13.0.10</version>
     </dependency>
     <dependency>
       <groupId>org.primefaces.extensions</groupId>
       <artifactId>primefaces-extensions</artifactId>
-      <version>14.0.10</version>
+      <version>13.0.14</version>
     </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>

--- a/deegree-services/deegree-webservices/pom.xml
+++ b/deegree-services/deegree-webservices/pom.xml
@@ -278,12 +278,12 @@
     <dependency>
       <groupId>org.primefaces</groupId>
       <artifactId>primefaces</artifactId>
-      <version>14.0.6</version>
+      <version>14.0.10</version>
     </dependency>
     <dependency>
       <groupId>org.primefaces.extensions</groupId>
       <artifactId>primefaces-extensions</artifactId>
-      <version>14.0.6</version>
+      <version>14.0.10</version>
     </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>

--- a/deegree-tools/deegree-tools-base/src/main/java/org/deegree/tools/coverage/gridifier/index/MultiLevelMemoryTileGridIndex.java
+++ b/deegree-tools/deegree-tools-base/src/main/java/org/deegree/tools/coverage/gridifier/index/MultiLevelMemoryTileGridIndex.java
@@ -50,9 +50,9 @@ import org.deegree.coverage.raster.geom.RasterGeoReference.OriginLocation;
 import org.deegree.geometry.Envelope;
 import org.deegree.geometry.GeometryFactory;
 import org.deegree.tools.coverage.gridifier.RasterLevel;
-import org.postgis.PGgeometry;
-import org.postgis.Point;
-import org.postgis.Polygon;
+import net.postgis.jdbc.PGgeometry;
+import net.postgis.jdbc.geometry.Point;
+import net.postgis.jdbc.geometry.Polygon;
 
 public class MultiLevelMemoryTileGridIndex implements MultiLevelRasterTileIndex {
 

--- a/deegree-tools/deegree-tools-base/src/main/java/org/deegree/tools/coverage/gridifier/index/PostGISRasterTileIndex.java
+++ b/deegree-tools/deegree-tools-base/src/main/java/org/deegree/tools/coverage/gridifier/index/PostGISRasterTileIndex.java
@@ -49,9 +49,9 @@ import org.deegree.coverage.raster.geom.RasterGeoReference.OriginLocation;
 import org.deegree.geometry.Envelope;
 import org.deegree.geometry.GeometryFactory;
 import org.deegree.tools.coverage.gridifier.RasterLevel;
-import org.postgis.PGgeometry;
-import org.postgis.Point;
-import org.postgis.Polygon;
+import net.postgis.jdbc.PGgeometry;
+import net.postgis.jdbc.geometry.Point;
+import net.postgis.jdbc.geometry.Polygon;
 
 public class PostGISRasterTileIndex implements MultiLevelRasterTileIndex {
 

--- a/pom.xml
+++ b/pom.xml
@@ -823,17 +823,17 @@
       <dependency>
         <groupId>net.postgis</groupId>
         <artifactId>postgis-jdbc</artifactId>
-        <version>2.5.1</version>
+        <version>2024.1.0</version>
       </dependency>
       <dependency>
         <groupId>net.postgis</groupId>
         <artifactId>postgis-geometry</artifactId>
-        <version>2.5.1</version>
+        <version>2024.1.0</version>
       </dependency>
       <dependency>
         <groupId>org.postgresql</groupId>
         <artifactId>postgresql</artifactId>
-        <version>42.7.3</version>
+        <version>42.7.4</version>
       </dependency>
       <!-- Oracle (Official provided driver from repo1.maven.org) -->
       <dependency>

